src/pytorch_lightning/profilers/hpu.py

```diff
@@ -0,0 +1,151 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import re
+from pathlib import Path
+from typing import Any, List, Optional, Union
+
+import torch
+
+from pytorch_lightning.profilers.pytorch import PyTorchProfiler
+from pytorch_lightning.trainer.connectors.data_connector import warning_cache
+from pytorch_lightning.utilities.imports import _KINETO_AVAILABLE
+from pytorch_lightning.utilities.rank_zero import rank_zero_warn
+
+if _KINETO_AVAILABLE:
+    from torch.profiler import profile, ProfilerActivity, tensorboard_trace_handler
+
+log = logging.getLogger(__name__)
+
+_PROFILER = profile
+
+
+class HPUProfiler(PyTorchProfiler):
```

Comment:
Hey @jerome-habana, great work there!

I wonder if we couldn't do slightly better there.

It seems the main reason for overriding was the error message. Maybe we could re-organize the PyTorch Profiler slightly to accommodate accelerator-specific activity better.

Reply:
Agree @tchaton . This would need some change to PT profiler with hpu support. We are working to enable some event mechanics there too.